src/actions/guilds/GUILD_DELETE.ts

```diff
@@ -1,20 +1,49 @@
-import { Action, Guild } from '@klasa/core';
+import { Action, ClientEvents, extender } from '@klasa/core';
 
 import type { GuildDeleteDispatch } from '@klasa/ws';
 
 export default class CoreAction extends Action {
 
-	public check(data: GuildDeleteDispatch): Guild | null {
-		return this.client.guilds.get(data.d.id) ?? null;
+	/**
+	 * Processes the event data from the websocket.
+	 * @since 0.0.1
+	 * @param data The raw data from {@link Client#ws}
+	 */
+	public run(data: GuildDeleteDispatch): void {
+		const guild = this.client.guilds.get(data.d.id);
+		if (!guild) {
+			if (!data.d.unavailable) return;
+
+			const created = new (extender.get('Guild'))(this.client, data.d, data.shard_id);
```

Comment:
This will create an entirely empty guild; intended?

Reply:
Yes, it is. In case a shard missed one guild or in the extreme edge case one guild comes out as unavailable, if we don't have it, we should cache it so GUILD_CREATE emits available instead of create.